openzwavemqtt/util.py

```diff
@@ -0,0 +1,209 @@
+"""Utility functions and classes for OpenZWave."""
+from typing import Any, Dict, List
+
+from .const import (
+    ATTR_LABEL,
+    ATTR_MAX,
+    ATTR_MIN,
+    ATTR_OPTIONS,
+    ATTR_PARAMETER,
+    ATTR_POSITION,
+    ATTR_TYPE,
+    ATTR_VALUE,
+    CommandClass,
+    ValueGenre,
+    ValueType,
+)
+from .exceptions import InvalidValueError, NotFoundError, WrongTypeError
+from .manager import OZWManager
+from .models.node import OZWNode
+
+
+def get_node_from_manager(
+    manager: OZWManager, instance_id: int, node_id: int
+) -> OZWNode:
+    """Get OZWNode from OZWManager."""
+    instance = manager.get_instance(instance_id)
+    if not instance:
+        raise NotFoundError(f"OZW Instance {instance_id} not found")
+
+    node = instance.get_node(node_id)
+    if not node:
+        raise NotFoundError(f"OZW Node {node_id} not found")
+
+    return node
+
+
+def set_config_parameter(node: OZWNode, parameter: int, new_value: Any) -> Any:
+    """Set config parameter to a node."""
+    value = node.get_value(CommandClass.CONFIGURATION, parameter)
+    if not value:
+        raise NotFoundError(
+            f"Configuration parameter {parameter} for OZW Node Instance not found"
+        )
+
+    # Bool can be passed in as string or bool
+    if value.type == ValueType.BOOL:
+        if isinstance(new_value, bool):
+            value.send_value(new_value)
+            return new_value
+        if isinstance(new_value, str):
+            if new_value.lower() in ("true", "false"):
+                payload = new_value.lower() == "true"
+                value.send_value(payload)
+                return payload
```

Comment:
Why are we returning the value we set?

Reply:
I suppose we don't need to. It's used in HA to log a message but we can use the original `selection` instead of the processed value

Reply:
Sometimes the selection and set value differ, right?

Reply:
in the case of a `ValueType.LIST`, the user can input the index or the label of the item and the code always uses the index. Other than that, the only transformation is casting to `int`/`bool`

Reply:
I'm ok with keeping the return value, if we think it's generally useful. It would be good to scope down the typing from `Any` if possible.